File: 05_roman_integer.js
function romanToInt(s) {
    const letters = s.replaceAll(" ", "").toLocaleUpperCase().split("");

    const c = {
        I: 1,
        V: 5,
        X: 10,
        L: 50,
        C: 100,
        D: 500,
        M: 1000,
    }

    let sum = 0

    for (let i = 0; i < letters.length; i++) {
        if (c[letters[i]] >= c[letters[i + 1]] || i === letters.length - 1) {
            sum += c[letters[i]]
        }
        else {
            sum -= c[letters[i]]
        }
    }
    
    console.log("result", sum)
    return sum
};

// romanToInt("XXIVV") // invalid
romanToInt("MCMXCIV") // 1994
romanToInt("MCDLXXVI") // 1476
romanToInt("XXXIV") // 34
romanToInt("XXIV") // 24
romanToInt("III") // 3
romanToInt("MCMXC") // 1990
romanToInt("CDXLIV") // 444
romanToInt("IX") // 9
romanToInt("IV") // 4

File: 05_roman_integer.rs
use std::collections::HashMap;

fn roman_to_int(s: &str) -> i32 {
    let mut c = HashMap::new();
    c.insert('I', 1);
    c.insert('V', 5);
    c.insert('X', 10);
    c.insert('L', 50);
    c.insert('C', 100);
    c.insert('D', 500);
    c.insert('M', 1000);

    let letters: Vec<char> = s.replace(" ", "").to_uppercase().chars().collect();
    let mut sum = 0;

    for i in 0..letters.len() - 1 {
        println!("loop");
        
        let current = c[&letters[i]];
        let next = c[&letters[i + 1]];

        if current >= next {
            sum += current;
        } else {
            sum -= current;
        }
    }
    sum
}

fn main() {
    let result = roman_to_int("MCDLXXVI");
    println!("Converted value: {}", result);
}